### ISSUE TITLE
A question about use pre-training code for other image shape

### ISSUE THREAD

**Issue by cypypccpy** [irom-lab/dppo#3]
Hi @allenzren ,
Thanks for the impressive work!
I'm trying to use the pretraining code on my custom datasets, but I see that this line of code (https://github.com/irom-lab/dppo/blob/main/model/common/vit.py#L83C62-L83C65) seems to only support 96x96 images. Can I directly modify it to support images of other shapes? Or is it currently limited to 96x96 images, otherwise it might cause other bugs?
Yuanpei

**Comment by allenzren** [comment]
Hi Yuanpei good to hear from you :)
Yes, you are right that the ViT is currently hard-coded to support 96x96 images as I borrowed the code from IBRL. I think we can definitely make it flexible. There shouldn't be other bug in the code if you use images of other shapes.
Do you have the time looking into this? I can implement the image size condition by tomorrow too.

**Comment by allenzren** [comment]
Also we need to check whether the ViT output fits the spatial embedding [here](https://github.com/irom-lab/dppo/blob/f5a8da57196a8341922bcead28a53d396dd2405b/model/common/modules.py#L10)

**Comment by cypypccpy** [comment]
Thank you for your response! Yeah, I've manually adjusted the code to support higher-dimensional image inputs, and I'm glad to hear that you plan to modify it (I'm concerned that my changes might cause errors hh:). 

I'm trying to train a pre-trained diffusion policy on real-world data using imitation learning, and then use DPPO for real-world RL fine-tuning, mainly for dexterous manipulation. So I wanted to ask a question—do you have evaluate the sample efficiency of DPPO during the fine-tuning phase, especially in comparison to some off-policy methods? This might have a significant impact on real-world RL hh.

**Comment by allenzren** [comment]
> Thank you for your response! Yeah, I've manually adjusted the code to support higher-dimensional image inputs, and I'm glad to hear that you plan to modify it (I'm concerned that my changes might cause errors hh:).

Oh cool that is great. Yea I don't think there will be other issue; but let me know if there is one

> do you have evaluate the sample efficiency of DPPO during the fine-tuning phase, especially in comparison to some off-policy methods?

Our experiments were focusing on comparing to (1) other policy parameterization also using PPO, and DPPO did better in sample efficiency and final performance, and (2) other diffusion-based RL methods (most of them are all pure off-policy), and the gain there from DPPO is more about training stability and thus able to tackle difficult tasks. We didn't compare DPPO with general non-diffusion, off-policy RL fine-tuning methods like RLPD, Cal-QL, and IBRL, but we are running the comparisons by next week or so.

I would be interesting to see how DPPO works in real-world fine-tuning. It would be great if you have some dense reward signal.

**Comment by cypypccpy** [comment]
Okay, understood, thank you for the detailed reply and suggestions! I will continue to follow.
